src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -78,7 +78,11 @@ protected function createController($controller)
      */
     protected function instantiateController($class)
     {
-        $controller = parent::instantiateController($class);
+        if ($this->container->has($class)) {
+            $controller = $this->container->get($class);
+        } else {
+            $controller = parent::instantiateController($class);
+        }
```

Comment:
I think we should make that check in `createController()` where we also retrieve the controller from the container when the `service_name:method_name` notation is used: https://github.com/dunglas/symfony/blob/invokable_service/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php#L56-L71